luigi/contrib/hive.py

```diff
@@ -180,8 +180,12 @@ class MetastoreClient(HiveClient):
     def table_location(self, table, database='default', partition=None):
         with HiveThriftContext() as client:
             if partition is not None:
-                partition_str = self.partition_spec(partition)
-                thrift_table = client.get_partition_by_name(database, table, partition_str)
+                try:
+                    import hive_metastore.ttypes
+                    partition_str = self.partition_spec(partition)
+                    thrift_table = client.get_partition_by_name(database, table, partition_str)
+                except hive_metastore.ttypes.NoSuchObjectException as e:
```

Comment:
You can also add test time dependencies in tox.ini:

https://github.com/spotify/luigi/blob/master/tox.ini